src/main/java/seedu/address/commons/function/ThrowingConsumer.java

```diff
@@ -0,0 +1,23 @@
+package seedu.address.commons.function;
+
+import java.util.function.Consumer;
+
+/**
+ * A Consumer that allows lambda expressions to throw exception.
+ * @param <T>
+ */
+@FunctionalInterface
+public interface ThrowingConsumer<T> extends Consumer<T> {
+
+    @Override
+    default void accept(final T elem) {
+        try {
+            acceptThrows(elem);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    void acceptThrows(T elem) throws Exception;
+
```

Comment:
this looks curious, how does this function work? why do you need it?

Reply:
When using lambda expressions, we need a functional interface to accept the arguments and return a result. For instance, if we want to iterate through a list of Persons:
```java
List<Person> persons = new List<>();
persons.forEach(person -> {
    //do something
});
```
The `person -> { //do something}` is passed into the`forEach` loop, which uses `java.util.function.Consumer` as its functional interface. `Consumer` accepts only one argument and returns void. 
However, by default, these functional interface doesn't throw exceptions. This means if you want to throw exceptions when using `forEach`, you have to create your own functional interface to throw it for you, or else the exception won't be handled.

[This post in stackoverflow](https://stackoverflow.com/questions/18198176/java-8-lambda-function-that-throws-exception/30246026#30246026) helped me gain some insights.

Reply:
`ThrowingConsumer` is used in `AddressBook.syncMasterPersonListWith(UniqueLifeInsuranceMap lifeInsuranceMap)` to handle `InsuranceNotFoundException`

Reply:
oh wow, this is really high level